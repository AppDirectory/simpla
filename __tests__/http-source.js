import HttpSource from '../src/http-source.js';

jest.unmock('../src/http-source.js');

describe('http-source', () => {
  let http;

  beforeAll(() => {
    http = new HttpSource({ data: '/content', uploads: '/uploads' });
  });

  afterEach(() => {
    fetch.resetMocks();
  });

  test('getting and transforming data', () => {
    fetch.mockResponseOnce(JSON.stringify({ text: 'Hello world [upload picture.png]' }));
    return http.get('/foo').then(response => {
      expect(fetch).toHaveBeenCalledWith('/content/foo.json');
      expect(response).toMatchSnapshot();
    });
  });

  test('dealing with a 404', () => {
    fetch.mockResponseOnce('', { status: 404 });
    return http.get('/foo').then(result => {
      expect(result).toBeNull();
    });
  });
});
