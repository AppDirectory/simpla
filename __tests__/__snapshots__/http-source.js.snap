// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`http-source getting and transforming data 1`] = `
Object {
  "text": "Hello world /uploads/picture.png",
}
`;
